q/unzip/unzip.q

```diff
@@ -0,0 +1,695 @@
+.finos.dep.include"../util/util.q"
+
+
+// Utilities
+
+// Read bytes from either a file or a byte vector.
+// @param x hsym or bytes
+// @param y offset
+// @param z length
+// @return z bytes from x, starting at y
+.finos.unzip.priv.bytes:{$[-11h=t:type x;read1(x;y;z);4h=t;x y+til z;'`type]}
+
+// Count bytes from either a file or a byte vector.
+// @param x hsym or bytes
+// @return count of bytes in x
+.finos.unzip.priv.bcount:{$[-11h=t:type x;hcount;count]x}
+
+// Split a subsection of data into fields.
+// Starts from offset and takes sum fields entries, splitting them according.
+//  to fields.
+// fields is a dictionary of field names and widths.
+// @param x fields
+// @param y offset
+// @param z data
+// @return the split subsection of the vector
+.finos.unzip.priv.split:{(key x)!(get sums prev x)cut z y+til sum x}
+
+// Parse byte(s) into a "number" (i.e. byte, short, int, or long, depending on the length).
+// @param x byte or bytes
+// @return byte, short, int, or long
+.finos.unzip.priv.parseNum:.finos.util.compose({$[1=count x;first;0x00 sv]x};reverse);
+
+// Parse byte(s) into bits; N.B. output is reversed to make flag dicts more natural.
+// @param x byte or bytes
+// @return bool vector
+.finos.unzip.priv.parseBits:.finos.util.compose(reverse;0b vs;.finos.unzip.priv.parseNum);
+
+// Parse bytes into a (global) unix timestamp.
+// @param x bytes
+// @return timestamp
+.finos.unzip.priv.parseUnixTime:.finos.util.compose(.finos.util.timestamp_from_epoch;.finos.unzip.priv.parseNum);
+
+// Parse a range of data with a header.
+// parser is a function of three arguments:
+//  Its first argument will be (data;extra); extra is passed as :: if not
+//   included.
+//  Its second argument will be the starting index of the record to extract.
+//  Its third argument will be the raw headers of the record, split and
+//   labeled according to fields.
+//  It should return (record;next index).
+// parser will be called until it returns next index equal to length.
+// @param x (parser;fields;extra)
```

Comment:
I think using multiple implicit parameters becomes confusing if the function is longer than a line or two.

Reply:
this entire file is written using implicit pa